apps/main/src/dex/components/PageCreatePool/hooks/useAutoDetectErc4626.ts

```diff
@@ -0,0 +1,92 @@
+import { useCallback, useState } from 'react'
+import { isAddress, isAddressEqual, type Address } from 'viem'
+import { NG_ASSET_TYPE } from '@/dex/components/PageCreatePool/constants'
+import { useIsErc4626 } from '@/dex/components/PageCreatePool/hooks/useIsErc4626'
+import { TokenId, TokenState, TokensInPoolState } from '@/dex/components/PageCreatePool/types'
+import { DEFAULT_ERC4626_STATUS } from '@/dex/store/createCreatePoolSlice'
+import useStore from '@/dex/store/useStore'
+
+type Candidate = {
+  tokenId: TokenId
+  address: Address
+}
+
+type UseAutoDetectErc4626Params = {
+  tokensInPool: TokensInPoolState
+}
+
+/**
+ * This hook is used to automatically detect if a token is an ERC4626 token and set the asset type to ERC4626
+ * @param tokensInPool - The tokens in the pool
+ * @returns A function to schedule a check for an ERC4626 token
+ */
+export const useAutoDetectErc4626 = ({ tokensInPool }: UseAutoDetectErc4626Params) => {
+  const updateNgAssetType = useStore((state) => state.createPool.updateNgAssetType)
+  const updateTokenErc4626Status = useStore((state) => state.createPool.updateTokenErc4626Status)
+  const [candidate, setCandidate] = useState<Candidate | null>(null)
+  const { isErc4626, isLoading, error, isSuccess } = useIsErc4626({ address: candidate?.address })
+
+  const setStatus = useCallback(
+    (tokenId: TokenId, status: TokenState['erc4626']) => {
+      updateTokenErc4626Status(tokenId, status)
+    },
+    [updateTokenErc4626Status],
+  )
+
+  const scheduleCheck = useCallback(
+    (tokenId: TokenId, address: Address) => {
+      setCandidate({ tokenId, address })
+      setStatus(tokenId, {
+        ...DEFAULT_ERC4626_STATUS,
+        isLoading: true,
+      })
+    },
+    [setStatus],
+  )
+
+  if (candidate) {
+    const currentToken = tokensInPool[candidate.tokenId]
+    if (!currentToken) {
+      setCandidate(null)
```

Comment:
never [call setState in the render loop](https://react.dev/reference/eslint-plugin-react-hooks/lints/set-state-in-render)

Reply:
Ugh, didn't know about this either. Such a shame we had to turn the linter rule off because the rule's broken in many other cases 😭